api/indexator.rb

```diff
@@ -0,0 +1,79 @@
+require 'date'
+require 'net/http'
+require 'json'
+
+class Indexator
+  attr_reader :contract_signature_date, :contract_start_date, :base_rent
+  BASE_POSSIBLE_YEARS = [1988, 1996, 2004, 2013].freeze
+
+  def initialize(contract_signature_date, contract_start_date, base_rent)
+    @contract_signature_date = contract_signature_date
+    @contract_start_date = contract_start_date
+    @base_rent = base_rent.to_i
+  end
+
+  def new_rent
+    base = define_base
+    base_month = define_base_month
+    current_month = self.current_month
+    base_index = self.base_index(base, base_month).to_f
+    current_index = self.current_index(base, current_month).to_f
+    (@base_rent * current_index / base_index).round(2)
```

Comment:
You can pass values directly to the method you call, no need to store them first in a variable, typically you could do:
```suggestion
    base_index = get_index(base, base_month).to_f
    current_index = get_index(base, current_month).to_f
    (@base_rent * current_index / base_index).round(2)
```
Furthermore, no need to use `self` keyword before the method calls (eg: `self.current_month` --> `current_month`)